elb_log_ingestor/elb_log_parse.py

```diff
@@ -0,0 +1,330 @@
+            "response": {"processing_time": match["response_processing_time"]},
+            "redirect_url": match["redirect_url"],
+            "sent_bytes": match["sent_bytes"],
+            "trace_id": match["trace_id"],
+            "target": {
+                "port": match["target_port"],
+                "processing_time": match["target_processing_time"],
+                "status_code": match["target_status_code"],
+                "ip": match["target_ip"],
+            },
+            "type": match["type"],
+            "request": {
+                "verb": match["request_verb"],
+                "url": match["request_url"],
+                "protocol": match["request_proto"],
+                "processing_time": match["request_processing_time"],
+                "creation_time": match["request_creation_time"],
+            },
+            "user_agent": match["user_agent"],
+        },
+    }
+    return new_match
+
+
+def format_elb_match(match: typing.Dict) -> typing.Dict:
+    """
+    Turn a match dict from an ALB log into a record appropriate for Elasticsearch
+    """
+    new_match = {
+        "@message": f"{match['request_verb'] or '-'} {match['request_url'] or '-'} {match['request_proto'] or '-'}",
+        "@elb": {
+            "response": {"processing_time": match["response_processing_time"]},
+            "elb": {"id": match["elb"], "status_code": match["elb_status_code"]},
+            "ssl": {"cipher": match["ssl_cipher"], "protocol": match["ssl_protocol"]},
+            "sent_bytes": match["sent_bytes"],
+            "target": {
+                "port": match["target_port"],
+                "processing_time": match["target_processing_time"],
+                "status_code": match["target_status_code"],
+                "ip": match["target_ip"],
+            },
+            "received_bytes": match["received_bytes"],
+            "request": {
+                "user_agent": match["user_agent"],
+                "url": match["request_url"],
+                "processing_time": match["request_processing_time"],
+                "verb": match["request_verb"],
+                "protocol": match["request_proto"],
+            },
+            "client": {"ip": match["client_ip"], "port": match["client_port"]},
+        },
+        "@timestamp": match["time"],
+    }
+    return new_match
+
+
+def add_metadata(record: typing.Dict, line: str, filename: str) -> typing.Dict:
+    """
+    Add common metadata to match _in place_
+    line: the line the match was based on
+    filename: the name of the logfile the log was found in
+    """
+    extra_metadata = {
+        "@input": "s3",
+        "@shipper.name": "elb_log_ingestor",
+        "@version": "1",
+        "@raw": line,
+        "@level": "INFO",
+        "tags": [],
+        "path": filename,
+    }
+    return {**record, **extra_metadata}
+
+
+def coerce_match_types(match: re.Match) -> typing.Dict:
+    """Convert an A/ELB log match into a dict with appropriate datatypes"""
```

Comment:
Format should match the others for consistency.

Reply:
(shrug is at Black, not at you 😄 )

Reply:
Funky...